src/app/static/view-question/view-question.component.html

```diff
@@ -2,14 +2,15 @@
   <app-question [question]="selectedQuestion"></app-question>
 </div>
 
-<br>
-<div class="msg-count"> total responses: {{totalMsgCount}}</div>
+<div class="msg-count"> 
+  total responses: {{selectedQuestion.responses.length}}
+</div>
 
 <div class="highlighted-response-component">
-  <app-highlighted-response *ngIf="highlightedResponse" [highlight] = "highlightedResponse"></app-highlighted-response>
+  <app-highlighted-response *ngIf="selectedQuestion.highlightedResponseId" [highnum] = "selectedQuestion.highlightedResponseId"></app-highlighted-response>
 </div>
 
-<div class="response-component" *ngFor="let response of questionResponses">
-  <app-response (highlighted)="refreshPage()" [response]="response"></app-response>
+<div class="response-component" *ngFor="let response of selectedQuestion.responses">
+  <app-response (highlighted)="ngOnChanges(null)" [response]="response"></app-response>
 </div>
-  <app-enter-response (newResponse)="refreshPage()" *ngIf="this.authService.account.expert"></app-enter-response>
+<app-enter-response (newResponse)="ngOnChanges(null)" *ngIf="this.authService.account.expert" [question]="selectedQuestion"></app-enter-response>
```

Comment:
I'm not sure this ngOnChanges(null) does anything?  ngOnChanges is the hook called when changes happen, not how we make the change detector run